packages/api/test/e2e/doughnut.e2e.ts

```diff
@@ -0,0 +1,206 @@
+                    console.log('\t', phase.toString(), `: ${section}.${method}`, data.toString());
+                });
+                if (transfer != undefined) {
+                    done();
+                } else {
+                    assert(true, 'false');
+                }
+            }
+        });
+    });
+
+    it('Fails when charlie uses bob\'s doughnut', async () => {
+        let doughnut = await makeDoughnut(
+            aliceKeyPair,
+            bob,
+            {'cennznet': makeCennznut('generic-asset', 'transfer')},
+        );
+        const doughnutB = new Doughnut(api.registry, doughnut);
+        const opt = {doughnut: doughnutB};
+        const tx = api.tx.genericAsset.transfer(16001, charlie.address, 10000);
+
+        await expect(tx.signAndSend(charlie, opt as any, () => {
+        })).rejects.toThrow('submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: {"Custom":180}');
+    });
+
+    it('fails when cennznut does not authorize the extrinsic method', async (done) => {
+        let doughnut = await makeDoughnut(
+            aliceKeyPair,
+            bob,
+            {'cennznet': makeCennznut('generic-asset', 'mint')},
+        );
+        const doughnutB = new Doughnut(api.registry, doughnut);
+        const opt = {doughnut: doughnutB};
+        const tx = api.tx.genericAsset.transfer(16001, charlie.address, 10000);
+
+        await tx.signAndSend(bob, opt as any, async ({events, status}) => {
+            if (status.isFinalized) {
+                const blockHash = status.value.toHex();
+                console.log('Completed at block hash', blockHash);
+                const allEvents = await api.query.system.events.at(blockHash);
+                const failed = allEvents.find(event => event.event.data.method === 'ExtrinsicFailed');
+                if (failed != undefined) {
+                    done();
+                } else {
+                    assert(false, 'expected extrinsic to fail');
+                }
+            }
+        });
+
+    });
+
+    it('fails when cennznut does not authorize the extrinsic module', async (done) => {
+        let doughnut = await makeDoughnut(
+            aliceKeyPair,
+            bob,
+            {'cennznet': makeCennznut('balance', 'transfer')},
+        );
+
+        const tx = api.tx.genericAsset.transfer(16001, charlie.address, 10000);
+        const doughnutB = new Doughnut(api.registry, doughnut);
+        const opt = {doughnut: doughnutB};
+
+        await tx.signAndSend(bob, opt as any, async ({events, status}) => {
+            if (status.isFinalized) {
+                const blockHash = status.value.toHex();
+                console.log('Completed at block hash', blockHash);
+                const allEvents = await api.query.system.events.at(blockHash);
+                const failed = allEvents.find(event => event.event.data.method === 'ExtrinsicFailed');
```

Comment:
Also the same suggestion as for line 170: checking other fields of the event for the failure's reason.

Reply:
The same here, first prove the created doughnut works fine, then use it in your test scenario.